setup.bzl

```diff
@@ -0,0 +1,149 @@
+    ],
+    visibility = ["//visibility:public"],
+)
+"""
+
+MARKUPSAFE_BUILD_FILE = """
+py_library(
+    name = "markupsafe",
+    srcs = glob(["markupsafe/*.py"]),
+    srcs_version = "PY2AND3",
+    visibility = ["//visibility:public"],
+)
+"""
+
+MISTUNE_BUILD_FILE = """
+py_library(
+    name = "mistune",
+    srcs = ["mistune.py"],
+    srcs_version = "PY2AND3",
+    visibility = ["//visibility:public"],
+)
+"""
+
+SIX_BUILD_FILE = """
+py_library(
+    name = "six",
+    srcs = ["six.py"],
+    srcs_version = "PY2AND3",
+    visibility = ["//visibility:public"],
+)
+"""
+
+def skydoc_repositories():
+    """Adds the external repositories used by the skylark rules."""
+    _include_if_not_defined(
+        git_repository,
+        name = "bazel_skylib",
+        remote = "https://github.com/bazelbuild/bazel-skylib.git",
+        tag = "0.6.0",
+    )
+    _include_if_not_defined(
+        git_repository,
+        name = "io_bazel_rules_sass",
+        remote = "https://github.com/bazelbuild/rules_sass.git",
+        tag = "1.15.1",
+    )
+    _include_if_not_defined(
+        git_repository,
+        name = "io_bazel_rules_sass",
+        remote = "https://github.com/bazelbuild/rules_sass.git",
+        tag = "1.15.1",
+    )
+    _include_if_not_defined(
+        git_repository,
+        name = "io_bazel",
+        remote = "https://github.com/bazelbuild/bazel.git",
+        # TODO: Update to a newer tagged version when available.
+        commit = "e7ebb7e68d35ae090d91fe6b4c92c1c831421faa",  # 2018-11-26
+    )
+    _include_if_not_defined(
+        git_repository,
+        name = "com_google_protobuf",
+        remote = "https://github.com/protocolbuffers/protobuf.git",
+        # Latest tagged version at time of writing is v3.6.1, which doesn't
+        # include fixes for --incompatible_package_name_is_a_function,
+        # --incompatible_new_actions_api, and possibly others.
+        # TODO: Update to a newer tagged version when available.
+        commit = "7b28271a61a3da0a37f6fda399b0c4c86464e5b3",  # 2018-11-16
+    )
+    _include_if_not_defined(
+        git_repository,
+        name = "io_bazel_rules_sass",
+        remote = "https://github.com/bazelbuild/rules_sass.git",
+        tag = "1.15.1",
+    )
+
+    # Protobuf expects an //external:python_headers label which would contain the
+    # Python headers if fast Python protos is enabled. Since we are not using fast
+    # Python protos, bind python_headers to a dummy target.
+    native.bind(
+        name = "python_headers",
+        actual = "//:dummy",
+    )
+
+    http_archive(
+        name = "markupsafe_archive",
+        urls = ["https://pypi.python.org/packages/source/M/MarkupSafe/MarkupSafe-0.23.tar.gz#md5=f5ab3deee4c37cd6a922fb81e730da6e"],
+        sha256 = "a4ec1aff59b95a14b45eb2e23761a0179e98319da5a7eb76b56ea8cdc7b871c3",
+        build_file_content = MARKUPSAFE_BUILD_FILE,
+        strip_prefix = "MarkupSafe-0.23",
+    )
+
+    native.bind(
```

Comment:
I know it's not new to this CL, but these binds are kind of unnecessary (this comment is not blocking but I don't think that's an option on github)

Reply:
They are? I'm not sure what you mean they are unnecessary (though admittedly I haven't tried removing them)

Reply:
Ok, unnecessary is an exaggeration, but it is just a convenience rename and I believe that there is a movement to get rid of bind (and in this case I think alias would work just as well)

Reply:
As discussed offline, it's pretty tricky removing these or replacing them with alias as-is :) Maybe another time